src/components/course/data/hooks.js

```diff
@@ -168,37 +187,54 @@ export function useCoursePriceForUserSubsidy(activeCourseRun, userSubsidy) {
         return null;
       }
 
-      const priceDetails = {
-        list: numberWithPrecision(listPrice),
+      const onlyListPrice = {
+        list: listPrice,
       };
-      if (!userSubsidy) {
-        return priceDetails;
-      }
 
-      const { discountType, discountValue } = userSubsidy;
-      let discountedPrice;
+      if (userSubsidyApplicableToCourse) {
+        const { discountType, discountValue } = userSubsidyApplicableToCourse;
+        let discountedPrice;
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
-        discountedPrice = listPrice - (listPrice * (discountValue / 100));
-      }
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
+          discountedPrice = listPrice - (listPrice * (discountValue / 100));
+        }
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
-        discountedPrice = Math.max(listPrice - discountValue, 0);
-      }
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
+          discountedPrice = Math.max(listPrice - discountValue, 0);
+        }
 
-      if (isDefinedAndNotNull(discountedPrice)) {
-        priceDetails.discounted = numberWithPrecision(discountedPrice);
-      } else {
-        priceDetails.discounted = priceDetails.list;
+        if (isDefinedAndNotNull(discountedPrice)) {
+          return {
+            ...onlyListPrice,
+            discounted: discountedPrice,
+          };
+        }
+        return {
+          ...onlyListPrice,
+          discounted: onlyListPrice.list,
+        };
       }
 
-      return priceDetails;
+      // Case 2: No subsidy, no offers for course
+      return onlyListPrice;
     },
-    [activeCourseRun, userSubsidy],
+    [activeCourseRun, userSubsidyApplicableToCourse],
   );
 
   return [coursePrice, currency];
-}
+};
+
+useCoursePriceForUserSubsidy.propTypes = {
+  activeCourseRun: PropTypes.shape({}).isRequired,
+  userSubsidyApplicableToCourse: PropTypes.shape({
+    discountType: PropTypes.string.isRequired,
+    discountValue: PropTypes.number.isRequired,
+    expirationDate: PropTypes.string.isRequired,
+    startDate: PropTypes.string.isRequired,
+    status: PropTypes.string.isRequired,
```

Comment:
nit: I believe the `status` key is missing from `userSubsidyApplicableToCourse` when it represents an offer; I think we can remove it from the propTypes here even though it exists when it represents a license. Should we make these consistently have the same data (i.e., make sure `userSubsidyApplicableToCourse` never contains `status`? I think we can assume that the data this contains is ready-for-use and will never need to read `status`.

Reply:
I was not aware of this difference `we can remove it from the propTypes here even though it exists when it represents a license.` So looks like I can clean it out of here.. we don't even use it?